fdbserver/DataDistribution.actor.cpp

```diff
@@ -146,10 +146,8 @@ void DataMove::validateShard(const DDShardInfo& shard, KeyRangeRef range, int pr
 		return;
 	}
 
-	if (!std::includes(
-	        this->primaryDest.begin(), this->primaryDest.end(), shard.primaryDest.begin(), shard.primaryDest.end()) ||
-	    !std::includes(
-	        this->remoteDest.begin(), this->remoteDest.end(), shard.remoteDest.begin(), shard.remoteDest.end())) {
+	if (!std::equal(this->primaryDest.begin(), this->primaryDest.end(), shard.primaryDest.begin()) ||
+	    !std::equal(this->remoteDest.begin(), this->remoteDest.end(), shard.remoteDest.begin())) {
```

Comment:
In the deleted condition, we care if this->primaryDest contains all elements of shard.primaryDest.
In the added condition, we care if shard.primaryDest contains all elements of this->primaryDest.
Do I understand correctly? If yes, could you explain why there is a change of the condition?

Reply:
Discussed offline.